Database/Beam/MySQL/Syntax.hs

```diff
@@ -508,6 +510,9 @@ instance HasSqlValueSyntax MysqlValueSyntax x => HasSqlValueSyntax MysqlValueSyn
     sqlValueSyntax Nothing = sqlValueSyntax SqlNull
     sqlValueSyntax (Just x) = sqlValueSyntax x
 
+instance HasSqlValueSyntax MysqlValueSyntax A.Value where
+    sqlValueSyntax = MysqlValueSyntax . escape . BL.toStrict . A.encode
```

Comment:
This is wrong. It produces the right output but isn't escaped/quoted correctly.

`MysqlValueSyntax . (\x -> emit "\"" <> x <> emit "\"") . escape . toStrict . encode` seems to work correctly, but that may be too much. I don't know if the `escape` is necessary here? Not sure what the security implications of unescaped SQL inside of JSON objects sent into the DB are

Reply:
`"INSERT INTO `saved_blueprints`(`saved_blueprint_id`, `user_id`, `name`, `currency`, `default_blueprint`, `charts`)VALUES (DEFAULT, 40240, 'WHAAAAT', 'JPY', FALSE, [1,2,3])"`

vs 

`"INSERT INTO `saved_blueprints`(`saved_blueprint_id`, `user_id`, `name`, `currency`, `default_blueprint`, `charts`)VALUES (DEFAULT, 40240, 'WHAAAAT', 'JPY', FALSE, \"[1,2,3]\")"` for comparison.

Reply:
For the input:

```json
{"mascot": "Our mascot is a dolphin named \"Sakila\"."}
```
,

```haskell
instance HasSqlValueSyntax MysqlValueSyntax Value where
  sqlValueSyntax v =
    MysqlValueSyntax $
    MysqlSyntax
      (\next doEscape before conn -> do
         escaped <- doEscape (toStrict $ encode v)
         next doEscape (before <> "'" <> byteString escaped <> "'") conn)
    -- MysqlValueSyntax .
    -- (\x -> emit "'" <> x <> emit "'") . escape . toStrict . encode
```
(trying to emulate how text does it, but it's the same as my commented attempt)
produces

```mysql
'{\\\"mascot\\\":\\\"Our mascot is a dolphin named \\\\\\\"Sakila\\\\\\\".\\\"}'
```

Which does work correctly in the DB, but according to [these docs]() should be:

```mysql
'{"mascot": "Our mascot is a dolphin named \\"Sakila\\"."}'
```